src/nvim/generators/gen_api_dispatch.lua

```diff
@@ -60,7 +60,6 @@ local function add_function(fn)
       fn.parameters[#fn.parameters] = nil
     end
     if #fn.parameters ~= 0 and fn.parameters[#fn.parameters][1] == 'arena' then
-      -- return value is allocated in an arena
       fn.arena_return = true
```

Comment:
So this means the `Arena` parameter is still supported? Should we mention that in the docstring for `FUNC_API_RET_ALLOC` or somewhere else? When would that be used?

Reply:
When it needs to be passed to another function.

Reply:
Ok so this is similar to the `channel_id` magic parameter. I guess we currently don't have a place where those are documented.

Reply:
This change removes the magic behavior associated with `arena` and *changes* it into an actually documented attribute instead. This PR literally is that documentation addition.

Reply:
My main question, which I still have, is: the name `arena_return` implies that this is a way to return stuff, i.e. an alternative to `FUNC_API_RET_ALLOC`. So does it make sense to mention `arena` parameter in the docstring on `FUNC_API_RET_ALLOC` ?

Reply:
ye, the name `arena_return` should be changed.